src/js/tests/vlmPipeline.test.js

```diff
@@ -0,0 +1,164 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+import { Tokenizer, VLMPipeline, DecodedResults, VLMDecodedResults } from "../dist/index.js";
+
+import assert from "node:assert";
+import { describe, it, before } from "node:test";
+import { promises as fs } from "node:fs";
+import { models } from "./models.js";
+import { createTestImageTensor, createTestVideoTensor } from "./utils.js";
+
+const MODEL_PATH = process.env.VLM_MODEL_PATH || `./tests/models/${models.VLLM.split("/")[1]}`;
+
+// Skip tests on macOS due to insufficient memory
+describe("VLMPipeline", { skip: process.platform === "darwin" }, () => {
+  let pipeline = null;
+
+  before(async () => {
+    try {
+      await fs.access(MODEL_PATH);
+    } catch {
+      console.log(`Model not found at ${MODEL_PATH}, skipping VLM tests`);
+      return;
+    }
+
+    pipeline = await VLMPipeline(MODEL_PATH, "CPU");
+    pipeline.setGenerationConfig({ max_new_tokens: 10 });
+  });
+
+  it("should generate text without images", async () => {
+    const result = await pipeline.generate("What is 2+2?");
+
+    assert.ok(result instanceof DecodedResults, "Result should be instance of DecodedResults");
+    assert.ok(
+      result instanceof VLMDecodedResults,
+      "Result should be instance of VLMDecodedResults",
+    );
+    assert.ok(result.texts.length > 0, "Should generate some output");
+  });
+
+  it("should generate text with images", async () => {
+    const testImage1 = createTestImageTensor();
+    const testImage2 = createTestImageTensor();
```

Comment:
I would create test images with different sizes (and videos if needed) outside the test case and reuse them in several `it(...)` test cases.